chronos/transports/stream.nim

```diff
@@ -700,12 +703,20 @@ when defined(windows):
         povl: RefCustomOverlapped
         proto: Protocol
 
-      var raddress = windowsAnyAddressFix(address)
+      var raddress = try: windowsAnyAddressFix(address)
+      except TransportAddressError as exc:
+        raiseAssert exc.msg
 
       toSAddr(raddress, saddr, slen)
       proto = Protocol.IPPROTO_TCP
-      sock = createAsyncSocket(raddress.getDomain(), SockType.SOCK_STREAM,
+      sock = try: createAsyncSocket(raddress.getDomain(), SockType.SOCK_STREAM,
```

Comment:
This procedure do not raise any exceptions except `OSError`.

Reply:
it depends on the platform you're compiling for: on windows it raises `OSError`, on linux it raises `IOSelectorException` and `ValueError` - this  is because it calls `register` which behaves differently